crates/noirc_evaluator/src/brillig/artifact.rs

```diff
@@ -10,6 +10,10 @@ pub(crate) type JumpLabel = usize;
 /// particular basic block starts.
 pub(crate) type BlockLabel = usize;
 
+/// Number of bytecodes in the prefix
+/// The prefix code is added at the start of every brillig function
+const PREFIX_LEN: usize = 3;
```

Comment:
Why is this needed?

Reply:
This is needed for failing the assert.
Instead of doing:
(0): (...)
(x): if assert_is_true, jump x+2
(x+1): trap
(x+2):...

we do:
(0): jump (3)
(1): trap
(2):stop
(3): (...)
(x): if assert_is_false, jump 1
(x+1): ....

We save one opcode for each assert.

Reply:
We can apply optimizations like this after the fact, if we decide to add further brillig passes. Lets get assert to work and we can open an issue about optimizing the bytecode

Reply:
I guess it's a question of what proper means, this definitely does not seem like the only possible jump location merging optimization

Reply:
To clarify, after the fact, was saying that we can apply optimizations to Brillig once we are done implementing. Whether we do it in-place or as a separate pass can be discussed when we are optimizing the bytecode.

The current implementation of Brillig generation should contain only the necessary items for generating Brillig